lib/trino-filesystem/src/main/java/io/trino/filesystem/fileio/ForwardingFileIo.java

```diff
@@ -61,4 +66,22 @@ public void deleteFile(String path)
             throw new UncheckedIOException("Failed to delete file: " + path, e);
         }
     }
+
+    @Override
+    public void deleteFiles(Iterable<String> pathsToDelete)
+            throws BulkDeletionFailureException
```

Comment:
This isn’t thrown by the method. Should we be using this instead?

Reply:
we cannot throw BulkDeletionFailureException becauase we cannot fill `BulkDeletionFailureException#numberFailedObjects`

see https://github.com/trinodb/trino/pull/15981#discussion_r1097111381 for more